tests/linearizability/failpoints.go

```diff
@@ -99,13 +103,92 @@ func (f killFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.Etcd
 		}
 	}
 
-	err := member.Start(ctx)
+	// get endpoints excluding the killed member client URLs
+	endpoints := make([]string, 0, len(clus.EndpointsV3()))
+	for _, ed := range clus.EndpointsV3() {
+		if ed != member.EndpointsV3()[0] {
+			endpoints = append(endpoints, ed)
+		}
+	}
+
+	if f.waitForSnapshot {
+		// wait for potential leader election
+		time.Sleep(2 * time.Second)
+		cc, err := clientv3.New(clientv3.Config{
+			Endpoints:            endpoints,
+			Logger:               zap.NewNop(),
+			DialKeepAliveTime:    1 * time.Millisecond,
+			DialKeepAliveTimeout: 5 * time.Millisecond,
+			Username:             rootUserName,
+			Password:             rootUserPassword,
+		})
+		require.NoError(t, err)
+		require.NoError(t, addTestUserAuth(ctx, cc))
+		// 5000 is the default number of snapshot count catch up entries
+		// DefaultSnapshotCatchUpEntries = 5000
```

Comment:
Do you have a production use case? 

I am supportive of making these available to e2e test though.

Reply:
I don't have prod case and the comment in config.go says
>// WARNING: only change this for tests. Always use "DefaultSnapshotCatchUpEntries"
On one hand, it's not a good idea to expose flag just because of testing, but doing 5k ops to trigger snapshot apply is also not great.

Another e2e test use case:
https://github.com/etcd-io/etcd/pull/15013#discussion_r1051683699

Reply:
I agree to make `DefaultSnapshotCatchUpEntries` configuable via a user-facing flag, two reasons:
1. It can benefit edge case, which usually is one-node cluster, and with limited resource (e.g memory). Setting a small value for `DefaultSnapshotCatchUpEntries` can reduce the memory usage. 
2. It can benefit the E2E test case.

Please let me know if any of you can do this, otherwise let me take care of it.

Reply:
Thanks @ahrtr I can take that as it looks like blocking my current PR. 

Because I just realize that linearizability test will repeat 60 times.. The current PR will add average 1min execution time per run. So It's not acceptable.  

https://github.com/etcd-io/etcd/blob/a59276c171d1647653b75679a12f85068d3b192b/.github/workflows/linearizability.yaml#L17

Reply:
ack. Will do on PST daytime and cc you and other stakeholders.